api/app/tests/apps/api/v1/test_private_departments.py

```diff
@@ -162,3 +162,37 @@ def test_delete_method_not_allowed(self):
 
         response = self.client.delete(self.detail_endpoint.format(pk=1))
         self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_SIG_2287(self):
+        # Connect a parent category to the Department so that we can check the URL generated for this category
+        self.category.departments.add(self.department, through_defaults={'is_responsible': True, 'can_view': True})
```

Comment:
Was already present in the code, but instances like these we should change `self.category` to `self.maincategory`. (Main and sub being special cases of the more general category. )

Having said that, this is no blocker. This is a naming problem in more parts of the SIA backend code.